api/src/tests/unit/controllers/AssetController.test.ts

```diff
@@ -38,35 +51,72 @@ describe('getAllAssets', () => {
     resetMockLogger(logger);
   });
 
-  it('Returns found asssets with a status code of 200', async () => {
+  it("Returns found asssets with a status code of 200", async () => {
     // Given
-    findAllMock.mockResolvedValue([testAsset]);
+    request.query = { page: "1", pageSize: "5" };
+    findAllMock.mockResolvedValue(
+      testPaginationAssets as PaginationResult<Asset>
+    );
 
     // When
     await assetController.getAllAssets(request, response, next);
 
     // Then
     expect(response.status).toHaveBeenCalledWith(200);
-    expect(response.send).toHaveBeenCalledWith([testAsset]);
+    expect(response.send).toHaveBeenCalledWith(testPaginationAssets);
     expect(response.end).toHaveBeenCalled();
     expect(next).not.toHaveBeenCalled();
     expect(logger.info).toHaveBeenCalledWith("Successfully retrieved Assets");
   });
 
-  it('Calls next middleware with an NotFoundError with message when no assets are found', async () => {
+  it("Calls next middleware with an NotFoundError with message when no assets are found", async () => {
     // Given
-    findAllMock.mockResolvedValue([]);
+    request.query = { page: "1", pageSize: "5" };
+    findAllMock.mockResolvedValue({
+      lastPage: 1,
+      hasMorePages: false,
+      totalRecords: 0,
+      data: [],
+    });
 
     // When
     await assetController.getAllAssets(request, response, next);
 
     // Then
-    expect(next).toHaveBeenCalledWith(ErrorController.NotFoundError("No Assets Found"));
+    expect(next).toHaveBeenCalledWith(
+      ErrorController.NotFoundError("No Assets Found")
+    );
     expectNonFinal(response);
   });
+
+  it.each([
+    [
+      { page: "0", pageSize: "1" },
+      { page: "1", pageSize: "0" },
+    ],
```

Comment:
S: Currently this is only running the test once, with two arguments. Each `object` should be wrapped in it's own array. W: Have the parameters of `page` and `pageSize` print out in the test name.

Reply:
Commit: 02dbeb7a0fa4367798ed906384e2bba6522c1763